src/main/java/com/example/java/maven/game2048/Game.java

```diff
@@ -7,9 +7,74 @@
     public Game(int winningNumber, Board board) {
         this.winningNumber = winningNumber;
         this.board = board;
+        board.placeTwoInitialElementsOnBoard();
+        MessagePrinter.printInstructionsBeforeGame();
     }
 
+
     public void start() {
+        while (!board.checkIfGameOver()) {
+            System.out.println(board);
+            makeValidMove();
+            if (board.checkIfWon(winningNumber)) {
```

Comment:
same here. Method `boolean playerWon()` should be in `Game` class. In `Board` you will have method: `boolean hasFieldWIthValue(int expectedValue)`

Reply:
Done